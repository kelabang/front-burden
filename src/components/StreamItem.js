/*
* @Author: Imam
* @Date:   2017-04-09 03:56:10
* @Last Modified by:   Imam
* @Last Modified time: 2017-04-09 04:33:51
*/

import React, { Component } from 'react';
import moment from 'moment';

class StreamItem extends Component {
	static propTypes = {
		username: React.PropTypes.string,
		userfullname: React.PropTypes.string,
		useravatar: React.PropTypes.string,
		value: React.PropTypes.string,
		datetime: React.PropTypes.string,
	}
	static defaultProps = {
		username: 'unknown',
		userfullname: 'unknown',
		useravatar: 'http://placehold.it/128x128',
		value: '',
		datetime: '0000-00-00 00:00:00',
	}
	relativeDateTime (date) {
		return moment(date, 'YYYY-MM-DD HH:mm:ss').fromNow();
	}
	render() {
		let relative = this.relativeDateTime(this.props.datetime);
		return (
			<article className="media">
				<div className="media-left">
					<figure className="image is-64x64">
						<img src={this.props.useravatar} alt="gambar" />
					</figure>
				</div>
				<div className="media-content">
					<div className="content">
						<p>
							<strong>{this.props.userfullname}</strong> <small>{'@' + this.props.username}</small> <small style={{float:'right'}}>{relative}</small>
							<br/>
							{this.props.value}
						</p>
					</div>
					<nav className="level">
						<div className="level-left">
							<a className="level-item inline-block">
								<span className="icon is-small"><i className="fa fa-reply"/></span>
							</a>
							<a className="level-item inline-block">
								<span className="icon is-small"><i className="fa fa-retweet"/></span>
							</a>
							<a className="level-item inline-block">
								<span className="icon is-small"><i className="fa fa-heart"/></span>
							</a>
						</div>
					</nav>
				</div>
			</article>
		);
	}
}

export default StreamItem;
