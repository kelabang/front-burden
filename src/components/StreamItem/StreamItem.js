/*
* @Author: Imam
* @Date:   2017-04-09 03:56:10
* @Last Modified by:   Imam
* @Last Modified time: 2017-04-09 14:11:15
*/

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

class StreamItem extends Component {
	static propTypes = {
		username: PropTypes.string,
		userfullname: PropTypes.string,
		useravatar: PropTypes.string,
		value: PropTypes.string,
		datetime: PropTypes.string,
	}
	static defaultProps = {
		username: 'unknown',
		userfullname: 'unknown',
		useravatar: 'http://placehold.it/128x128',
		value: '',
		datetime: '0000-00-00 00:00:00',
	}
	relativeDateTime (date) {
		return moment(date, 'YYYY-MM-DD HH:mm:ss').fromNow();
	}
	render() {
		let relative = this.relativeDateTime(this.props.datetime);
		return (
			<article className="media">
				<div className="media-left">
					<figure className="image is-64x64">
						<img src={this.props.useravatar} alt="gambar" />
					</figure>
				</div>
				<div className="media-content">
					<div className="content">
						<p>
							<strong className="fullname">{this.props.userfullname}</strong> <small className="username">{'@' + this.props.username}</small> <small className="datetime" style={{float:'right'}}>{relative}</small>
							<br/>
							{this.props.value}
						</p>
					</div>
					<nav className="level">
						<div className="level-left">
							<a className="level-item inline-block">
								<span className="icon is-small"><i className="fa fa-reply"/></span>
							</a>
							<a className="level-item inline-block">
								<span className="icon is-small"><i className="fa fa-retweet"/></span>
							</a>
							<a className="level-item inline-block">
								<span className="icon is-small"><i className="fa fa-heart"/></span>
							</a>
						</div>
					</nav>
				</div>
			</article>
		);
	}
}

export default StreamItem;
