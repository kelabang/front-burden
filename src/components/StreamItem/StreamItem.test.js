import React from 'react';
import ReactDOM from 'react-dom';
// import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import StreamItem from './StreamItem.js';


it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<StreamItem />, div);
});

const mockup = {
	'username': 'ma4m',
	'userfullname': 'imam tauhid',
	'useravatar': 'http://placehold.it/128x128',
	'value': 'hello world',
	'datetime': '1991-11-29 00:00:00'
};

test('StreamItem displayed data', () => {
	const component = renderer.create(
		<StreamItem username={mockup.username} userfullname={mockup.userfullname} useravatar={mockup.useravatar} value={mockup.value} datetime={mockup.datetime} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
