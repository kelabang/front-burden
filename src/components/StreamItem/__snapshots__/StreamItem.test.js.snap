exports[`test StreamItem displayed data 1`] = `
<article
  className="media">
  <div
    className="media-left">
    <figure
      className="image is-64x64">
      <img
        alt="gambar"
        src="http://placehold.it/128x128" />
    </figure>
  </div>
  <div
    className="media-content">
    <div
      className="content">
      <p>
        <strong
          className="fullname">
          imam tauhid
        </strong>
         
        <small
          className="username">
          @ma4m
        </small>
         
        <small
          className="datetime"
          style={
            Object {
              "float": "right",
            }
          }>
          25 years ago
        </small>
        <br />
        hello world
      </p>
    </div>
    <nav
      className="level">
      <div
        className="level-left">
        <a
          className="level-item inline-block">
          <span
            className="icon is-small">
            <i
              className="fa fa-reply" />
          </span>
        </a>
        <a
          className="level-item inline-block">
          <span
            className="icon is-small">
            <i
              className="fa fa-retweet" />
          </span>
        </a>
        <a
          className="level-item inline-block">
          <span
            className="icon is-small">
            <i
              className="fa fa-heart" />
          </span>
        </a>
      </div>
    </nav>
  </div>
</article>
`;
